src/main/java/com/autilus/hw1006/LinkedListImpl.java

```diff
@@ -0,0 +1,138 @@
+package com.autilus.hw1006;
+
+import java.util.NoSuchElementException;
+
+public class LinkedListImpl<T> implements List<T> {
+    Node<T> first;
+    Node<T> last;
+    private int size = 0;
+
+    private void checkIndex(int index) {
+        if (index < 0 || index > size) throw new NoSuchElementException();
+    }
+
+    private Node<T> getNodeByIndex(int index) {
+        checkIndex(index + 1);
+        Node<T> node;
+        if (index < size << 1) {
+            node = first;
+            for (int i = 0; i < index; i++) {
+                node = node.getNext();
+            }
+        } else {
+            node = last;
+            for (int i = size; i > index; i--) {
+                node = node.getPrev();
+            }
+        }
+        return node;
+    }
+
+    private Node<T> getNodeByValue(T value) {
+        Node<T> node = first;
+        for (int i = 0; i < size; i++) {
+            if (node.getValue().equals(value)) {
+                break;
+            } else {
+                node = node.getNext();
+            }
+        }
+        return node;
+    }
+
+    @Override
+    public void add(T value) {
+        Node<T> node = new Node<>(last, value, null);
+        if (isEmpty()) {
+            first = node;
```

Comment:
add last = node;

Reply:
I add "_last = node;_" below, because I need to write it in "_if_ " and "_else_"